pkg/segment/reader/segread/segreader.go

```diff
@@ -457,6 +477,35 @@ func (sfr *SegmentFileReader) deToResults(results map[uint16]map[string]interfac
 	return true
 }
 
+func (sfr *SegmentFileReader) deToResultsCnameBased(results map[string][]utils.CValueEnclosure,
+	orderedRecNums []uint16) bool {
+
+	for recIdx, rn := range orderedRecNums {
+		dwIdx := sfr.deRecToTlv[rn]
+		dWord := sfr.deTlv[dwIdx]
+
+		if dWord[0] == utils.VALTYPE_ENC_SMALL_STRING[0] {
+			results[sfr.ColName][recIdx].CVal = string(dWord[3:])
+			results[sfr.ColName][recIdx].Dtype = utils.SS_DT_STRING
+		} else if dWord[0] == utils.VALTYPE_ENC_BOOL[0] {
+			results[sfr.ColName][recIdx].CVal = toputils.BytesToBoolLittleEndian(dWord[1:])
+			results[sfr.ColName][recIdx].Dtype = utils.SS_DT_BOOL
+		} else if dWord[0] == utils.VALTYPE_ENC_INT64[0] {
+			results[sfr.ColName][recIdx].CVal = toputils.BytesToInt64LittleEndian(dWord[1:])
+			results[sfr.ColName][recIdx].Dtype = utils.SS_DT_SIGNED_NUM
+		} else if dWord[0] == utils.VALTYPE_ENC_FLOAT64[0] {
+			results[sfr.ColName][recIdx].CVal = toputils.BytesToFloat64LittleEndian(dWord[1:])
+			results[sfr.ColName][recIdx].Dtype = utils.SS_DT_FLOAT
+		} else if dWord[0] == utils.VALTYPE_ENC_BACKFILL[0] {
+			results[sfr.ColName][recIdx].Dtype = utils.SS_DT_BACKFILL
+		} else {
+			log.Errorf("SegmentFileReader.deToResults: de only supported for str/int64/float64/bool")
+			return false
+		}
```

Comment:
nit: log the `dWord[0]` value. Also a `switch` might be cleaner

Reply:
addressed in the latest commit.